### PR TITLE
xtensa-build-all.sh: Default private key leaking between platforms

### PR DESCRIPTION
**Pull request by softwarecki**
Private key file passed by user is stored in PRIVATE_KEY_OPTION variable. When building for many platforms, if user not pass path to key file, default value will be set in PRIVATE_KEY_OPTION. During building second platform, default key from the first one are treated as key passed by user and default key will not be applied.

I introduced the new variable PLATFORM_PRIVATE_KEY, which holds default signing key for current platform. If user passed own key, this value is later set in this variable. The xtensa-build-zephyr script works the same way. Changed DEFAULT_PLATFORMS and moved TGL platform to the middle of the list to make sure any future leakage like this one is detected immediately.

### CODE REVIEW

**Review by marc-hb — APPROVED** [review]
Important fix, thank you!

The checkpatch warnings in https://github.com/thesofproject/sof/runs/5426929673?check_suite_focus=true are either wrong or not worth fixing.

sof-ci/jenkins is down for the week-end, will have to wait Monday.

**Review by lgirdwood — APPROVED** [review]

**Review by marcinszkudlinski — APPROVED** [review]

File: scripts/xtensa-build-all.sh [hunk 1]
```diff
@@ -330,12 +321,12 @@ do
...
-                        HOST="xtensa-mt8186-elf"
-                        XTENSA_TOOLS_VERSION="RI-2020.5-linux"
-                        ;;
+		mt8186)
+			PLATFORM="mt8186"
+			XTENSA_CORE="hifi5_7stg_I64D128"
+			HOST="xtensa-mt8186-elf"
+			XTENSA_TOOLS_VERSION="RI-2020.5-linux"
+			;;
 		mt8195)
```
Comment (dbaluta) [inline scripts/xtensa-build-all.sh:330]: what is this change here? It doesn't say anything about it in the commit message? If this is indeed a change, can you please add it in a separate commit and explained what happened. I cannot figure it out from github diff. Maybe you changed some tabs? :)

File: scripts/xtensa-build-all.sh [hunk 2]
```diff
@@ -7,8 +7,8 @@ set -e
 
 # Platforms with a toolchain available in the latest Docker image and
 # built by the -a option.
-DEFAULT_PLATFORMS=(  byt cht bdw hsw apl skl kbl cnl sue icl jsl \
-                    imx8 imx8x imx8m imx8ulp tgl tgl-h rn mt8186 mt8195 )
+DEFAULT_PLATFORMS=(  byt cht bdw hsw tgl tgl-h apl skl kbl cnl sue icl jsl \
+                    imx8 imx8x imx8m imx8ulp rn mt8186 mt8195 )
```
Comment (dbaluta) [inline scripts/xtensa-build-all.sh:12]: This is unrelated to commit message description. I see you reordered some platforms? Please move this in a separate commit with proper explanation.

it doesn't look like an alphabetical reorder.

File: scripts/xtensa-build-all.sh (hunk 1 above)
Comment (marc-hb) [reply scripts/xtensa-build-all.sh:330]: This is the message of the second commit:
> [xtensa-build-all.sh: Code style correction](https://github.com/thesofproject/sof/pull/5475/commits/0affed20e9d99ced951a2fe31c79521ebfa75135) 
>
> Combined case label tgl and tgl-h similarly as in xtensa-build-zephyr.sh
> Corrected code formatting.

@dbaluta the Github description / "cover letter" and the git commit messages are different.

File: scripts/xtensa-build-all.sh (hunk 2 above)
Comment (dbaluta) [reply scripts/xtensa-build-all.sh:12]: github description will be lost after this is submitted to sof repo. Lets add every useful information inside the git commit messages.

Comment (marc-hb) [reply scripts/xtensa-build-all.sh:12]: Agreed. I used to copy/paste/diverge commit messages into the github description. I stopped when there is more than one commit; now I'm only providing a very short summary and pointing at the commit messages.

Comment (softwarecki) [reply scripts/xtensa-build-all.sh:12]: This change doesn't seem so important as to separate it into a separate commit. Besides, its justification would require a reference to the commit in which it is currently located.

**Comment by softwarecki** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
CI showing known and unrelated errors for SDW PM.

File: scripts/xtensa-build-all.sh [hunk 3]
```diff
@@ -274,29 +275,14 @@ do
 			XTENSA_TOOLS_VERSION="RG-2017.8-linux"
 			HAVE_ROM='yes'
 			;;
-		tgl)
+		tgl|tgl-h)
 			PLATFORM="tgplp"
```
Comment (marc-hb) [inline scripts/xtensa-build-all.sh:279]: This changed PLATFORM for `tgl-h`. Revert submitted in #5495
